xlators/mgmt/glusterd/src/glusterd-handler.c

```diff
@@ -2653,14 +2653,19 @@ glusterd_peer_hostname_update(glusterd_peerinfo_t *peerinfo,
     GF_ASSERT(peerinfo);
     GF_ASSERT(hostname);
 
-    ret = gd_add_address_to_peer(peerinfo, hostname);
+    ret = gd_add_address_to_peer_head(peerinfo, hostname);
     if (ret) {
         gf_msg(THIS->name, GF_LOG_ERROR, 0,
                GD_MSG_HOSTNAME_ADD_TO_PEERLIST_FAIL,
                "Couldn't add address to the peer info");
         goto out;
     }
 
+    /* Also set peerinfo->hostname to the first address */
+    if (peerinfo->hostname != NULL)
+        GF_FREE(peerinfo->hostname);
+    peerinfo->hostname = gf_strdup(hostname);
```

Comment:
In theory at least, gf_strdump might fail due to lack of memory. In Gluster we usually check for such issues.

Reply:
@mykaul 
Thanks for your review.
Here, the code and comment are referenced to existing code in gd_update_peerinfo_from_dict(). 
And, peerinfo->hostname itself can be NULL. If gf_strdup failed, it should be a not fatal. Log and return the error，I think maybe no help to this abnormal condition.
What dou you think about it?

Reply:
@mykaul

Reply:
Look at other cases of gf_strdup() - I think it's worthwhile logging a warning, as otherwise it'll be very difficult to track such an issue. If it's not fatal, logging should be enough.

Reply:
@mykaul  
There are a lot of gf_strdup(hostname) calls. I think it’s better to do this check in another patch.

Reply:
I will do this in another patch, later..